### DISCUSSION TITLE
Robust image date of creation fallback mechanism

### DISCUSSION THREAD

**Discussion by maciejmatczak** [tfeldmann/organize#405] [Q&A]
Properly detecting date time of image taken is pretty challenging from what I can tell. Modification time is mostly bad, sometimes exif data doesn't exist, sometimes date cannot be recognized from file name... but combining all of above, can render well working strategy.

How to implement a config that would try following strategies and fallback in case of failure to the next one:

- recognize from name (regex)
- use exif.exif.datetimeoriginal
- use exif.image.datetime.year

Creating each one, separate is simple, but in combination... I have no clue :)

**Comment by bjeans** [comment]
It looks like this comment might be similar to what you are looking for https://github.com/tfeldmann/organize/discussions/116#discussioncomment-7703824
